### PR TITLE
17  authorization header (master)

### PR DESCRIPTION
**Pull request by jasoncoposky**

### CODE REVIEW

**Comment by trel** [comment]
seeing four commits here
- two without issue number (connection pool)
- do you want to squash them?

File: irods_specific_implementation/indexed_connection_pool_with_expiry.hpp [hunk 1]
```diff
@@ -89,11 +91,28 @@ namespace irods {
 
     struct connection_context
     {
-        bool                      in_use;
+	    bool                      in_use;
+	    bool                      evict_immediately;
```
Comment (trel) [inline irods_specific_implementation/indexed_connection_pool_with_expiry.hpp:95]: indentation

File: irods_specific_implementation/indexed_connection_pool_with_expiry.hpp [hunk 2]
```diff
@@ -89,11 +91,28 @@ namespace irods {
...
+        {
+            // ctor
+        }
     }; // connection_context
 
+    namespace {
+	    auto now_in_seconds() -> time_type
+	    {
+		    return std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now());
+	    } // now_in_seconds
```
Comment (trel) [inline irods_specific_implementation/indexed_connection_pool_with_expiry.hpp:113]: indentation

File: irods_specific_implementation/irods_rest_auth_api_implementation.h [hunk 3]
```diff
@@ -5,31 +5,118 @@
...
+				64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64
+			};
+
+			auto in_len = _in.size();
+			if (in_len % 4 != 0) {
+                THROW(SYS_INVALID_INPUT_PARAM,
+                      "input is not a multiple of 4");
+            }
+
+			auto out_len = in_len / 4 * 3;
```
Comment (trel) [inline irods_specific_implementation/irods_rest_auth_api_implementation.h:53]: indentation

Comment (jasoncoposky) [reply irods_specific_implementation/irods_rest_auth_api_implementation.h:53]: looks correct to me in the file, perhaps a tab/space issue in github?

Comment (jasoncoposky) [reply irods_specific_implementation/irods_rest_auth_api_implementation.h]: ran :retab on the file, should be fixed.

File: irods_specific_implementation/indexed_connection_pool_with_expiry.hpp (hunk 1 above)
Comment (trel) [reply irods_specific_implementation/indexed_connection_pool_with_expiry.hpp:95]: still see this one?  definitely has a leading tab in there

File: irods_specific_implementation/indexed_connection_pool_with_expiry.hpp (hunk 2 above)
Comment (trel) [reply irods_specific_implementation/indexed_connection_pool_with_expiry.hpp:113]: same here

Comment (jasoncoposky) [reply irods_specific_implementation/indexed_connection_pool_with_expiry.hpp]: ran :retab on that file as well

**Comment by trel** [comment]
ready for #

**Pull request merged**